src/main/java/task/TaskList.java

```diff
@@ -0,0 +1,155 @@
+package task;
+
+import java.lang.reflect.Array;
+import java.util.ArrayList;
+import parser.Parser;
+
+public class TaskList {
+    public static final String TODO_TASK = "T";
+    public static final String DEADLINE_TASK = "D";
+    public static final String EVENT_TASK = "E";
+
+    private static ArrayList<Task> tasks;
+    private static int taskCount;
+
+    public TaskList() {
+        tasks = new ArrayList<>();
+        taskCount = 0;
+    }
+
+    public static int getLength() {
+        return tasks.size();
+    }
+
+    public static ArrayList<Task> getTasks() {
+        return tasks;
+    }
+
+    public static void incrementTaskCount() {
+        taskCount += 1;
+    }
+
+    public void displayTasksList() {
+        if (tasks.isEmpty()) {
+            System.out.println("\tYour list does not contain any tasks.");
+        } else {
+            int taskNumber = 1;
+            System.out.println("\tHere are the tasks in your list:");
+            for (Task task : tasks) {
+                System.out.println("\t" + taskNumber + "." + task.getTaskDetails());
+                taskNumber += 1;
+            }
+        }
+        System.out.println();
+    }
+
+    public void markTask(int taskIndex) throws IndexOutOfBoundsException {
+        try {
+            tasks.get(taskIndex).markAsDone();
+        } catch (IndexOutOfBoundsException e) {
+            System.out.println("Invalid task number. Please enter a valid task number");
+        }
+    }
+
+    public void unmarkTask(int taskIndex) throws IndexOutOfBoundsException {
+        try {
+            this.tasks.get(taskIndex).unmarkAsDone();
+        } catch (IndexOutOfBoundsException e) {
+            System.out.println("Invalid task number. Please enter a valid task number\n");
+        }
+    }
+
+    public void addTask(String taskType, String[] userInput) {
+        if (taskType.equals(Parser.TODO_COMMAND)) {
+            handleToDo(userInput);
+        } else if (taskType.equals(Parser.DEADLINE_COMMAND)) {
+            handleDeadline(userInput);
+        } else {
+            handleEvent(userInput);
+        }
```

Comment:
This is not a good practice, the else case should only be for the cases when a match is not found. It should not assume that if it is not either then it is else. So for making code more robust:

```suggestion
        if (taskType.equals(Parser.TODO_COMMAND)) {
            handleToDo(userInput);
        } else if (taskType.equals(Parser.DEADLINE_COMMAND)) {
            handleDeadline(userInput);
        } else if (taskType.equals(Parser.EVENT_COMMAND)) {
            handleEvent(userInput);
        } else {
            throw InvalidCommandException() // Some Exception you deem right
        }
```

Reference: https://nus-cs2113-ay2324s2.github.io/website/se-book-adapted/chapters/codeQuality.html#use-the-default-branch